plugins/studymaster-web/src/main/java/de/dfki/vsm/xtension/studymasterWeb/WebStudyMasterExecutor.java

```diff
@@ -0,0 +1,306 @@
+    private static final String sREQUEST_RESULT_VAR = "request_result_var" ;
+    private static final String sREQUEST_RESULT_VAR_DEFAULT = "request_result" ;
+
+    /** Project variable set when the remote Web GUI connects via websocket. */
+    private static final String sGUI_CONNECTED_VAR = "gui_connected_var" ;
+    private static final String sGUI_CONNECTED_VAR_DEFAULT = "gui_connected" ;
+
+    /** HTTP port */
+    private static final String sJAVALIN_PORT_VAR = "port" ;
+    private static final String sJAVALIN_PORT_VAR_DEFAULT = "8080" ;
+
+
+    /** The singleton logger instance.*/
+    private final LOGConsoleLogger mLogger = LOGConsoleLogger.getInstance();
+    /** The list of websocket connections. Yes, it will be possible to have several GUIs on the same scene flow.*/
+    private final ArrayList<WsConnectContext> mWebsockets = new ArrayList<>();
+    /** The Javalin HTTP server. */
+    private Javalin mHttpServer;
+
+    private String mRequestResultVar ;
+
+    private String mSceneflowGoVar ;
+
+    /** This is the name of a Boolean global project variable that will be bet to _true_ when at least one web app is connected.*/
+    public String mGUIConnectedVar = "";
+
+    /** This is a "cache" of the last request sent to all connected GUIs.
+     * The string contains the message already formatted to the communication protocol.
+     * It will be used to send again the request to GUIs connecting after the REQUEST action was issued.
+     * If null, no pending request is active.
+     */
+    private String mLastRequestMessage = null;
+
+    /**
+     * Default constructor, pass values to superclass.
+     *
+     * @param config  Plugin configuration, as it can be set via the extensions settings, or manually via xml.
+     * @param project The running project the plugin is applied to.
+     */
+    public WebStudyMasterExecutor(PluginConfig config, RunTimeProject project) {
+        super(config, project);
+    }
+
+
+    @Override
+    public synchronized String marker(long id) {
+        return "$(" + id + ")";
+    }
+
+
+    @Override
+    public void launch() {
+        mLogger.message("Loading StudyMaster message sender and receiver ...");
+
+        //
+        // Retrieve property values
+        mGUIConnectedVar = mConfig.getProperty(sGUI_CONNECTED_VAR, sGUI_CONNECTED_VAR_DEFAULT);
+        final int http_port = Integer.parseInt(Objects.requireNonNull(mConfig.getProperty(sJAVALIN_PORT_VAR, sJAVALIN_PORT_VAR_DEFAULT)));
+        mRequestResultVar = mConfig.getProperty(sREQUEST_RESULT_VAR, sREQUEST_RESULT_VAR_DEFAULT) ;
+        mSceneflowGoVar = mConfig.getProperty(sGO_VAR, sGO_VAR_DEFAULT);
+
+        // Start the HTTP server
+        mHttpServer = Javalin.create(config -> {
+            config.addStaticFiles("/react-studymaster/build");
+            config.enforceSsl = true;
+        }).start(http_port);
+
+        // Set callbacks to manage WebSocket events
+        mHttpServer.ws("/ws", ws -> {
```

Comment:
maybe we should not use /ws for every plugin, so we can reuse the web server at some point

Reply:
I see, but at the moment we can leave it like this. We are anyway distinguishing plugins by port.
In future, we can have indeed a centralised Javalin in the core.